### PR TITLE
SQL update 2018-11-16

### PR DESCRIPTION
**Pull request by kbevers**
As promised, a new update to the SQL. There's some minor changes to the model, otherwise the big change is that a bunch of constraints has been added.

### CODE REVIEW

**Comment by bjornharrtell** [comment]
Because we had to remove empty newlines for it to work to load with sqlplus this PR has conflicts. Probably easiest to use your version and remove empty newlines again. If possible future updates should avoid empty newlines.

File: test/fixtures/sql/fikspunkt_forvaltning.sql [hunk 1]
```diff
@@ -290,8 +331,394 @@ COMMENT ON COLUMN SAGSEVENT.REGISTRERINGTIL IS 'Tidspunktet hvor en ny registrer
...
+NOPARALLEL;
+
+ALTER TABLE PUNKT ADD (
+  CONSTRAINT PUNKT_R01
+  UNIQUE (ID)
+  USING INDEX ID_IDX_0001
+  ENABLE VALIDATE);
+
+
+ALTER TABLE PUNKT ADD (
```
Comment (AsgerPetersen) [inline test/fixtures/sql/fikspunkt_forvaltning.sql]: Er denne constraint ikke duplikeret eller er jeg blind?

**Comment by kbevers** [comment]
> If possible future updates should avoid empty newlines.

Sounds like a bad idea to me. Humans will have to understand this, so I would prefer not removing newlines that improves readability.

File: test/fixtures/sql/fikspunkt_forvaltning.sql (hunk 1 above)
Comment (kbevers) [reply test/fixtures/sql/fikspunkt_forvaltning.sql]: Det ser sådan ud, ja. Jeg melder den lige tilbage til Steen der har lavet disse constraints. Godt spottet.

**Comment by AsgerPetersen** [comment]
> > If possible future updates should avoid empty newlines.
> 
> Sounds like a bad idea to me. Humans will have to understand this, so I would prefer not removing newlines that improves readability.

It is only **empty lines** inside CREATE TABLE statements that are bad.

**Comment by bjornharrtell** [comment]
As of #14 we can now load the schema as is without removal of newlines or whatnot but there appears to be issues with the schema itself, at least according to sqlplus. This is a partial capture of the output:

```
UNIQUE (ID)
  *
ERROR at line 3:
ORA-02261: such unique or primary key already exists in the table


ALTER TABLE OBSERVATION ADD
*
ERROR at line 8:
ORA-01735: invalid ALTER TABLE option


SP2-0734: unknown command beginning "IF :new.RE..." - rest of line ignored.
SP2-0734: unknown command beginning "IF :new.SA..." - rest of line ignored.
SP2-0042: unknown command "end" - rest of line ignored.
ALTER TABLE OBSERVATION ADD
*
ERROR at line 8:
ORA-01735: invalid ALTER TABLE option



Trigger created.


Trigger created.


Warning: Trigger created with compilation errors.


Trigger created.


Trigger created.


Trigger created.


Index created.

  PRIMARY KEY
  *
ERROR at line 2:
ORA-02260: table can have only one primary key


ALTER TABLE OBSERVATIONTYPE ADD
*
ERROR at line 15:
ORA-01735: invalid ALTER TABLE option


VALUES ('Koteforskel fra fikspunkt1 til fikspunkt2 (h2-h1) opmålt trigonometrisk' , 'trigonometrisk_koteforskel', TRUE, FALSE, 'Koteforskel [m]', 'Nivellementslængde [m]', 'Antal opstillinger', 'Afstandsafhængig varians pr. målt koteforskel [m^2/m^2]', 'Afstandsuafhængig varians pr. målt koteforskel [m^2]', 'Total varians koteforskel [m^2]', NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL)
                                                                                                                             *
ERROR at line 2:
ORA-00984: column not allowed here


VALUES ('Horisontal retning med uret fra opstilling til sigtepunkt (reduceret til ellipsoiden)' , 'retning', TRUE, FALSE, 'Retning [m]', 'Varians  retning hidrørende instrument, pr. sats  [rad^2]', 'Samlet centreringsvarians for instrument prisme [m^2]', 'Total varians retning [rad^2]', NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL)
                                                                                                                   *
ERROR at line 2:
ORA-00984: column not allowed here


VALUES ('Horisontal afstand mellem opstilling og sigtepunkt (reduceret til ellipsoiden)' , 'horisontalafstand', TRUE, FALSE, 'Afstand [m]', 'Afstandsafhængig varians afstandsmåler [m^2/m^2]', 'Samlet varians for centrering af instrument og prisme, samt grundfejl på afstandsmåler [m^2]', 'Total varians horisontalafstand [m^2]', NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL)
                                                                                                                      *
ERROR at line 2:
ORA-00984: column not allowed here


VALUES ('Skråafstand mellem opstilling og sigtepunkt' , 'skråafstand', TRUE, FALSE, 'Afstand [m]', 'Afstandsafhængig varians afstandsmåler pr. måling [m^2/m^2]', 'Samlet varians for centrering af instrument og prisme, samt grundfejl på afstandsmåler pr. måling [m^2]', 'Total varians skråafstand [m^2]', NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL)
                                                                                       *
ERROR at line 2:
ORA-00984: column not allowed here


VALUES ('Zenitvinkel mellem opstilling og sigtepunkt' , 'zenitvinkel', TRUE, FALSE, 'Zenitvinkel [rad]', 'Instrumenthøjde [m]', 'Højde sigtepunkt [m]', 'Varians zenitvinkel hidrørende instrument, pr. sats  [rad^2]', 'Samlet varians instrumenthøjde/højde sigtepunkt [m^2]', 'Total varians zenitvinkel [rad^2]', NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL)
                                                                             *
ERROR at line 2:
ORA-00984: column not allowed here


VALUES ('Vektor der beskriver koordinatforskellen fra punkt 1 til punkt 1 (v2-v1)' , 'vektor', TRUE, FALSE, 'dx [m]', 'dy [m]', 'dz [m]', 'Afstandsafhængig varians [m^2/m^2]', 'Samlet varians for centrering af antenner [m^2]', 'Total varians [m^2]', 'Varians dx [m^2]', 'Varians dy [m^2]', 'Varians dz [m^2]', 'Covarians dx, dy [m^2]', 'Covarians dx, dz [m^2]', 'Covarians dy, dz [m^2]', NULL, NULL, NULL)
                                                                                                     *
ERROR at line 2:
ORA-00984: column not allowed here
```

**Comment by kbevers** [comment]
There's apparently a bunch if errors in the constraints and the insert statements at the bottom of the file. At this time I think it is best to just discard those. Accordring to Nikolaj, everything after line 342 should be considered from this version of the DDL. That is what he needed to do to get his migration scripts running.

**Comment by bjornharrtell** [comment]
Replaced by https://github.com/Septima/fikspunktsregister/pull/27.

**Pull request closed without merging**